common/src/jni/main/cpp/conscrypt/native_crypto.cc

```diff
@@ -3599,6 +3599,141 @@ static jint evp_aead_ctx_op(JNIEnv* env, jlong evpAeadRef, jbyteArray keyArray,
     return static_cast<jint>(actualOutLength);
 }
 
+
+//doggo
+static jint evp_aead_ctx_op_buf(JNIEnv* env, jlong evpAeadRef, jbyteArray keyArray, jint tagLen,
+                            jobject outBuffer, jbyteArray nonceArray,
+                            jobject inBuffer, jbyteArray aadArray,
+                            evp_aead_ctx_op_func realFunc) {
+
+    const EVP_AEAD* evpAead = reinterpret_cast<const EVP_AEAD*>(evpAeadRef);
+    JNI_TRACE("evp_aead_ctx_op(%p, %p, %d, %p, %p, %p, %p)", evpAead, keyArray, tagLen,
+              outBuffer, nonceArray, inBuffer, aadArray);
+
+    jclass bytebuffer_class = env->FindClass("java/nio/ByteBuffer");
```

Comment:
AFAIK, `env->FindClass` and subsequent `GetMethodID`s are pretty expensive to call so please considering to get them beforehand once.